docs/book/src/developer/core/controllers/machine.md

```diff
@@ -1,20 +1,39 @@
-# Machine  Controller
+# Machine Controller
 
-![](../../../images/cluster-admission-machine-controller.png)
+The Machine controller is responsible for reconciling the Machine resource.
+
+In order to allow Machine provisioning on different type of infrastructure, The Machine resource references
+an Machine object, e.g. AWSMachine, GCMachine etc.
+
+The [InfraMachine resource contract](../../providers/contracts/infra-machine.md) defines a set of rules a provider is expected to comply in order to allow
+the expected interactions with the Machine controller.
+
+Among those rules:
+- InfraMachine MUST report a [provider ID](../../providers/contracts/infra-machine.md#inframachine-provider-id) for the Machine
+- InfraMachine SHOULD define a [failure domain](../../providers/contracts/infra-machine.md#inframachine-failure-domain) where machines should be placed in
```

Comment:
```suggestion
- InfraMachine SHOULD define a [failure domain](../../providers/contracts/infra-machine.md#inframachine-failure-domain) field that decides where machines are placed in
```
Maybe. Sould define a failureDomain sounds a bit unclear

Otherwise it sounds like the InfraMachine comes up with the failureDomain, but I think it's the other way around

Reply:
Rephrased, PTAL

Reply:
If what I'm proposing is ok,  I'm going to open an issue to track this + brings this up in the office hours

Reply:
In general sounds good to me.

I assume we still have use cases going forward where no failureDomain is specified in Machine, then the infra provider just picks one and we want to surface that back?

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/11232